compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkiaLayer.skiko.kt

```diff
@@ -186,27 +192,26 @@ internal class SkiaLayer(
         invalidate()
     }
 
-    // TODO(demin): support perspective projection for rotationX/rotationY (as in Android)
-    // TODO(njawad) Add camera distance leveraging Sk3DView along with rotationX/rotationY
-    // see https://cs.android.com/search?q=RenderProperties.cpp&sq= updateMatrix method
-    // for how 3d transformations along with camera distance are applied. b/173402019
     private fun updateMatrix() {
         val pivotX = transformOrigin.pivotFractionX * size.width
         val pivotY = transformOrigin.pivotFractionY * size.height
 
         matrix.reset()
+        matrix.translate(x = -pivotX, y = -pivotY)
         matrix *= Matrix().apply {
-            translate(x = -pivotX, y = -pivotY)
-        }
-        matrix *= Matrix().apply {
-            translate(translationX, translationY)
-            rotateX(rotationX)
-            rotateY(rotationY)
             rotateZ(rotationZ)
+            rotateY(rotationY)
+            rotateX(rotationX)
             scale(scaleX, scaleY)
         }
         matrix *= Matrix().apply {
-            translate(x = pivotX, y = pivotY)
+            // the camera location is passed in inches, set in pt
+            val depth = cameraDistance * 72f
+            set(2, 3, -1f / depth)
+            set(2, 2, 0f)
```

Comment:
```suggestion
            set(row = 2, column = 3, v = -1f / depth)
            set(row = 2, column = 2, v = 0f)
```
Maybe with argument names it will be more understandable?